### PR TITLE
limit send buffer size

### LINKED ISSUE #137: Limit send buffer size
Like the receiver buffer, we need to have a configurable limit of how many packets (or bytes?) to have in the send buffer before dropping them on the floor. 

Part of this is adding this to the tokio interface by adding backpressure (not letting someone add a packet to the buffer when its full)

### PR DESCRIPTION
**Pull request by robertream**
Closes #137

### CODE REVIEW

File: srt-protocol/src/settings/connection.rs [hunk 1]
```diff
@@ -59,6 +61,7 @@ impl From<options::SocketOptions> for ConnInitSettings {
             bandwidth: options.sender.bandwidth,
             statistics_interval: options.session.statistics_interval,
             recv_buffer_size: options.receiver.buffer_size / options.session.max_segment_size,
+            send_buffer_size: options.sender.buffer_size / options.session.max_segment_size,
```
Comment (russelltg) [inline srt-protocol/src/settings/connection.rs:64]: Is buffer_size supposed to be how many data payload bytes? If so it should be `buffer_size / (mss - headers)`

**Comment by russelltg** [comment]
overall looks great. just the one clarification

File: srt-protocol/src/settings/connection.rs (hunk 1 above)
Comment (robertream) [reply srt-protocol/src/settings/connection.rs:64]: Good point, I checked the options documentation and the receive buffer size limit should also be calculated this way too.


> SRTO_RCVBUF
> 
> Receive Buffer Size, in bytes. Note, however, that the internal setting of this value is in the number of buffers, each one of size equal to SRT payload size, which is the value of SRTO_MSS decreased by UDP and SRT header sizes (28 and 16). The value set here will be effectively aligned to the multiple of payload size.

**Comment by robertream** [comment]
@russelltg ok, I also fixed the max receive buffer size calculation too

**Pull request merged**